processor/src/test/java/org/mapstruct/ap/test/value/SpecialOrderMapper.java

```diff
@@ -0,0 +1,48 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.value;
+
+import org.mapstruct.InheritInverseConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.ValueMapping;
+import org.mapstruct.ValueMappingType;
+import org.mapstruct.ValueMappings;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Sjaak Derksen
+ */
+@Mapper
+public interface SpecialOrderMapper {
+
+    SpecialOrderMapper INSTANCE = Mappers.getMapper( SpecialOrderMapper.class );
+
+    OrderDto orderEntityToDto(OrderEntity order);
+
+    @ValueMappings({
+        @ValueMapping( target = "DEFAULT", valueMappingType = ValueMappingType.NULL ),
```

Comment:
So I have to admit I find the approach around `ValueMappingType` less comprehensible that what we used to have. It would be so much nicer to read (and write) like so:

    @ValueMapping( target = "DEFAULT", source="null" )
    @ValueMapping( target = "null", source = "STANDARD" )

I understand that the proposed mechanism is more versatile, but I feel we are adding complexity without gaining very much at this point. The only thing supported is enum mappings, but it's happening in a much more concealed way now by this generic approach.

My preference is to introduce an explicit `@EnumMapping`, because that's what we want out if this right now. We still can add `@ValueMapping` further down the road if we see the need for other value mappings.